### ISSUE TITLE
Became unresponsive

### ISSUE THREAD

**Issue by dawnerd** [docker/for-mac#442]
### Expected behavior
Not hang

### Actual behavior
Hangs

### Information
Restarted docker and issue persists.

Diagnostic ID: 08ADA003-4317-49EF-BEAA-581A2AC502D8
Docker for Mac: 1.12.0-a (Build 11213)
macOS: Version 10.11.6 (Build 15G31)
[ERROR]  docker-cli
         docker ps failed
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux

### Steps to reproduce

Was running a bunch of builds that included npm installs / grunt tasks.

**Comment by mchiang0610** [comment]
Merging with #119

**Comment by samoht** [comment]
Detected symptom of problem 'invariant-violated' in 08ADA003-4317-49EF-BEAA-581A2AC502D8/20160828-001100.

An invariant within the code of one of the components was not held. Docker needs to be restarted.

May be related to docker/for-mac#89

The following log matches:
```
Aug 26 20:10:17 goatbook Docker[com.docker.driver.amd64-linux][59786] <Notice>: VM: 2016-08-27 03:10:17 +0000 UTC daemon.info vsudd: Connection 273 to: 2376 from: 00000000
Aug 26 20:10:17 goatbook Docker[com.docker.driver.amd64-linux][59786] <Notice>: VM: 2016-08-27 03:10:17 +0000 UTC daemon.info vsudd: 273 Done. read: 254 written: 4549
Aug 26 20:10:17 goatbook Docker[com.docker.osxfs][59783] <Notice>: Volume.stop e228688b1b26263136b4233c25b9622aa0fc0dbd33693b2c88a389cef03a0f35 (paths = [/Users/troywhiteley/Projects/inspired-survey])
Aug 26 20:10:17 goatbook Docker[com.docker.osxfs][59783] <Error>: WATCH SUBSCRIBER REMOVAL INVARIANT VIOLATED
Aug 26 20:10:17 goatbook Docker[com.docker.driver.amd64-linux][59786] <Notice>: VM: 2016-08-27 03:10:17 +0000 UTC daemon.info vsudd: 259 Done. read: 310 written: 2573014
Aug 26 20:10:17 goatbook Docker[com.docker.driver.amd64-linux][59786] <Notice>: VM: 2016-08-27 03:10:17 +0000 UTC daemon.info vsudd: Connection 274 to: 2376 from: 00000000
Aug 26 20:10:17 goatbook Docker[com.docker.driver.amd64-linux][59786] <Notice>: VM: 2016-08-27 03:10:17 +0000 UTC daemon.info vsudd: Connection 275 to: 2376 from: 00000000
```

/cc @dsheets

**Comment by dsheets** [comment]
This invariant violation was a symptom of a use-after-free bug in the file system event watching code which occurred due to multiple containers with volume mounts racing to set up their mounts. It is possible that it caused your hangs or crashes and has been fixed in the recently released Beta 29.3. Please open a new issue for any further problems. Thanks!
